### PR TITLE
Fel Juggernaut Death Bug

### LINKED ISSUE #459: Destroying Fel Horde power generators should destroy their associated Juggernauts

### PR DESCRIPTION
**Pull request by Sharkmike**
Implemented the ability for Fel Juggernauts to die when their partner Power Generators die.

Closes #459

### CODE REVIEW

**Comment by Sharkmike** [comment]
Close #459 ticket

File: src/WarcraftLegacies.Source/Mechanics/Fel Horde/FelHordeJuggernautDeath.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+using MacroTools.Wrappers;
+using System;
+using WCSharp.Shared.Data;
+using static War3Api.Common;
+
+namespace WarcraftLegacies.Source.Mechanics.Fel_Horde
+{
+
+ 
+  internal class FelHordeJuggernautDeath
```
Comment (YakaryBovine) [inline src/WarcraftLegacies.Source/Mechanics/Fel Horde/FelHordeJuggernautDeath.cs:11]: Mark this as `static` as it only has static methods.

File: src/WarcraftLegacies.Source/Mechanics/Fel Horde/FelHordeJuggernautDeath.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+﻿using MacroTools;
+using MacroTools.Wrappers;
+using System;
+using WCSharp.Shared.Data;
+using static War3Api.Common;
+
+namespace WarcraftLegacies.Source.Mechanics.Fel_Horde
+{
+
+
```
Comment (YakaryBovine) [inline src/WarcraftLegacies.Source/Mechanics/Fel Horde/FelHordeJuggernautDeath.cs:10]: Maybe add a little XML comment here describing what it's doing.

File: src/WarcraftLegacies.Source/Mechanics/Fel Horde/FelHordeJuggernautDeath.cs (hunk 1 above)
Comment (YakaryBovine) [inline src/WarcraftLegacies.Source/Mechanics/Fel Horde/FelHordeJuggernautDeath.cs:11]: I think we should rename this to just `JuggernautDeath`. It's already in the Fel Horde folder so the specificity is redundant.

**Review by YakaryBovine — CHANGES_REQUESTED** [review]
Well done.

**Comment by Sharkmike** [comment]
Pushed with changes made as requested.

**Review by YakaryBovine — APPROVED** [review]

**Review by YakaryBovine — APPROVED** [review]

**Pull request merged**